### ISSUE TITLE
Dark Souls 3 (374320)

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/Proton#6309]
# Compatibility Report
Dark Souls 3 (374320)

## System Information
- GPU: RX570
- Driver/LLVM version: Mesa 22.1.7
- Kernel version: Linux fedora 6.0.5-200.fc36.x86_64
- Link to full system information report: [Report](https://gist.github.com/Vovkiv/f16fee8abdaa8086d3345fe2aa289b6b)
- Proton version: experimental-7.0-20221102

## I confirm:
- [*] that I haven't found an existing compatibility report for this game.
- [*] that I have checked whether there are updates for my system available.

[Logs](https://gist.github.com/Vovkiv/458196a43190000ed3a52575cb980151)

## Symptoms
Video demonstration: https://youtu.be/tn_wpcguOJU
Character cannot move, use items, roll, attack, anything. Only use left stick/WASD to rotate character and right stick/mouse to move camera, open inventory and try use items from there, options. 

## Reproduction
Fedora Silverblue 36 (all installed updates), Flatpak Steam version from Flathub(1.0.0.75, com.valvesoftware.Steam), Installed package `steam-devices`, Wayland
Controller is Dualshock 4

1. On Fedora silverblue, open Steam installed from Flatpak
2. Open Dark Souls 3 with latest Proton Experimental
3. Start new game (or load existing save)
4. Cannot do anything

What i tried:
1. Switching proton versions. Same behaviour for 7-0.4, 6.3-8, 5.13-6
2. Switching from Wayland to X11
3. Check files integrity. All files was alright.
4. Disconnecting Dualshock 4 and use only keyboard
5. Reinstall game

There was no such problems in DS3 on Fedora Workstation 36 with dnf steam on same system.
Also worth noting, that on current system other fromsoftware games works without such problems: Dark Souls 2 (335300), Dark Souls 1 Remastered (570940), Sekiro (814380), Elden Ring (1245620). And since this games (At least, Dark Souls series) should be build on same engine and they works without such issue, i'm not sure if this problem exclusive to Fedora Silverblue&Flatpak Steam.

UPD:
Try this: https://github.com/ValveSoftware/Proton/issues/6309#issuecomment-1595762247; it seems fixed issue, so check it out.

**Comment by deathxxx123** [comment]
This is familiar issue, but i forgot what was the fix. Just dont play DS3 from long time and do not remember.
I think it was something with controller, but there are a lot people that suffer from this issue.
Just look into google.
That is happen on Linux and Win.

**Comment by ghost** [comment]
I've had similar issues with essentially all of the souls games before and the only thing that consistently fixed it aside from randomly tinfoiling around, was to disable steam input for the game and load the gamepad preset in the desktop configuration (make sure dualshock support is activated in the big picture menu). Also try using https://github.com/GloriousEggroll/proton-ge-custom
I hope I could help you

**Comment by ghost** [comment]
> I've had similar issues with essentially all of the souls games before and the only thing that consistently fixed it aside from randomly tinfoiling around, was to disable steam input for the game and load the gamepad preset in the desktop configuration (make sure dualshock support is activated in the big picture menu). Also try using https://github.com/GloriousEggroll/proton-ge-custom I hope I could help you

If you look at video that i posted, you can see, that character doesn't play any animation: when rotating, when changing weapons, when trying to use items (even from inventory), when trying to attack, when trying to move, when trying to make gesture. Game registers that i tried to make gesture but nothing happened.


It doesn't matter if steam input is on or off, it doesn't matter if gamepad is connected, it doesn't matter if i trying to use keyboard instead. Game registers input just fine, steam input or not, gamepad or keyboard + mouse, but the problem is that something *broken* "inside", something that prevents character animations, movement, etc.


So i more then enough sure, that this problem has nothing to do with input but rather proton/wine, game itself of even fedora silverblue/ flatpak steam

**Comment by XenBad** [comment]
I have the same issue. I'm on Fedora Silverblue 36 and I'm using the Steam flatpak. 

I've tried all of the suggested fixes that I could find, including: 
- Uninstalling the game and re-installing it in a new Steam library
- Disabling Steam input
- Using the Steam RPM package
- Using proton-ge
- Using only a keyboard
- Verifying integrity
- Starting a new game

I have no issues with Sekiro as well.

**Comment by ghost** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6309#issuecomment-1322766506

So, you also cannot move, use items, gestures, etc?
If true, that raises question if it related to fedora silverblue, rather then flatpak steam

**Comment by XenBad** [comment]
> > Replying to [#6309 (comment)](https://github.com/ValveSoftware/Proton/issues/6309#issuecomment-1322766506)
> 
> So, you also cannot move, use items, gestures, etc? If true, that raises question if it related to fedora silverblue, rather then flatpak steam

My issue is exactly the same as the one in the video.  I tried using Steam as an RPM and the issue still occurred, so it seems more likely that it is Fedora Silverblue related.

Here is my [system information report](https://github.com/ValveSoftware/Proton/files/10069976/system-information.txt) if that helps.

**Comment by ghost** [comment]
> Here is my [system information report](https://github.com/ValveSoftware/Proton/files/10069976/system-information.txt) if that helps.

Well, from glancing it, it seems, that systems are similar in hardware and software sense, except major difference is GPU, but overall that pretty similar.

And let me guess, you also tried different versions of proton/wine/etc and it also have no difference?
Then it's highly possible that this is silverblue issues, since no such problems was in Workstation.

Inputs clearly works, doesn't matter what you use: gamepad or k&m, or something else. The best way to debug this issue, it seems, to use debug mods for ds3 (which, i believe, whould be on nexusmods or gamebanana). 

Character *refuse* to play animation, and this is probably the root of problem (or, at least, i think). I'm not big game development expert, but i believe, everything in character functionality tied to animations.
To prove my point:
* To change weapon, character need to play animation when they hide old weapon and when they take weapon out of "pockets". When you try to switch it with d-pad, nothing is happening and weapon will never change because animation didn't start at first place.
* But you can change weapon through inventory, which will work because this method doesn't require any animation to start/finish.
* You can change items in quickslots with d-pad since they also doesn't require animations.
* When you change direction in which character face, they should make animation when they turn around and move legs (resulting in very small, but moving from initial place), but here character turn around without any animations and without moving even for 1 pixel, so i think moving also tied to animations.

So my conclusion is: game fails to play animations for character that player controls, effectively render game unplayable at all. What cause this... i don't.

**Comment by XenBad** [comment]
@Vovkiv I fixed my issue, but I don't know if it will work for you. I created a new steam library folder inside of `~/.var/app/com.valvesoftware.Steam` and moved Dark Souls 3 into it. I tried this solution before on my old PC, and it didn't work, this may be due to where I created the directory, or something else, I'm not sure.

**Comment by ghost** [comment]
> @Vovkiv I fixed my issue, but I don't know if it will work for you. I created a new steam library folder inside of `~/.var/app/com.valvesoftware.Steam` and moved Dark Souls 3 into it. I tried this solution before on my old PC, and it didn't work, this may be due to where I created the directory, or something else, I'm not sure.

It seems, that it didn't change anything for me

**Comment by TimLovesTech** [comment]
> This is familiar issue, but i forgot what was the fix. Just dont play DS3 from long time and do not remember. I think it was something with controller, but there are a lot people that suffer from this issue. Just look into google. That is happen on Linux and Win.

I would agree it is something controller related, or at least how Steam is presenting the controller to the game. I had this same issue when trying to play Elden Ring in the beginning. My setup is completely different from OP, so Flatpak Steam is basically the common denominator. I messed around with a lot of things, and even switch to a wired Xbox controller (was originally trying a Stadia controller and wireless Xbox). In changing controllers I also added a kernel module to the mix for the controller and controls in Elden Ring worked. I have since started playing with a wireless Xbox controller and the issue never came back.

Current System Info:
Display Server: X11
GPU: NVIDIA RTX 3600 Ti
Driver version: 525.47.04 (Vulkan Dev)
Kernel version: Linux Arch 6.1.4-273-tkg-pds
Proton version: GE-Proton7-43

**Comment by ghost** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6309#issuecomment-1374900828

I think it might be separate issue, not related here, since even only keyboard don't work, no matter if steam input is on or off, or there is gamepad connected at all. Really strange issue. Maybe if i will have time, it worth check it rpm steam working here.

**Comment by ghost** [comment]
Apparently, issue not related to flatpak at all? I deleted flatpak steam and installed rpm steam, installed fresh copy of game and get same issue.

**Comment by ghost** [comment]
The game no longer launches for me since Proton 8. Almost gets to the loading screen then crashes to desktop. Going back to 7 solves it
Log https://pastebin.com/JtE1dRYL

**Comment by grimshins** [comment]
I can also confirm that the game fails to launch with Proton 8, and that reverting to Proton 7 works. This is true of Dark Souls 2: Scholar of the First Sin and Dark Souls Remastered as well, although they at least make it to the title screen, and only crash upon loading a game.

**Comment by mbriar** [comment]
crashes for me too on Proton 8, log: 
[steam-374320.log](https://github.com/ValveSoftware/Proton/files/11260764/steam-374320.log)

**Comment by grimshins** [comment]
Today's update of Proton 8 (8.0-1d) resolved the crashing issue in the Dark Souls games for me.

**Comment by ghost** [comment]
Just want to point out (yet again):
I installed Fedora 38 Silverblue that was released on 19 april, with same setup and hardware: Flatpak steam with installed steam-devices and dualshock 4. And my theory about broken animations seems more realistic. On last game session on Fedora 37 Workstation (where game works fine), when character was in death animation i quit game. And in Souls games, game remember that player in death animation state, so when you load game again, game will play death animation as it supposed to, "You Died" text will appear and then game will reload. But the thing is: When i loaded in game, i was on 0 hp, character was was standing on default staying animation. And nothing happens after that. If someone care, i can record it, but result same as in opening issue recording: Anything tied to animations (walking, running, attacking, etc) won't work because animations themself won't play.

**Comment by mionciao** [comment]
You need to install Dark Souls 3 in a shorter path, worked for playing on Manjaro.

Read from this: https://steamcommunity.com/app/374320/discussions/0/365163686082978255/

**Comment by ghost** [comment]
> You need to install Dark Souls 3 in a shorter path, worked for playing on Manjaro.
> 
> Read from this: https://steamcommunity.com/app/374320/discussions/0/365163686082978255/

how long your current path to game? @mionciao

**Comment by hebrewd** [comment]
I'm facing the same issue, I haven't played for a while and then for a savefile where I already had progress, I couldn't move.
I tried to reinstall in various different location, including the one that already work.

I am using f2fs on arch however, perhaps the file system is also problematic?

**Comment by ghost** [comment]
> I'm facing the same issue, I haven't played for a while and then for a savefile where I already had progress, I couldn't move. I tried to reinstall in various different location, including the one that already work.
> 
> I am using f2fs on arch however, perhaps the file system is also problematic?

@hebrewd i don't think it's issue with filesystem. I played game on btfrs, ext4 and same issues. The thing is, this happens only on Fedora Silverblue, regular Fedora Workstation doesn't have this problem. It doesn't matter if steam is in flatpak or rpm.

**Comment by hebrewd** [comment]
> > I'm facing the same issue, I haven't played for a while and then for a savefile where I already had progress, I couldn't move. I tried to reinstall in various different location, including the one that already work.
> > I am using f2fs on arch however, perhaps the file system is also problematic?
> 
> @hebrewd i don't think it's issue with filesystem. I played game on btfrs, ext4 and same issues. The thing is, this happens only on Fedora Silverblue, regular Fedora Workstation doesn't have this problem. It doesn't matter if steam is in flatpak or rpm.

@Vovkiv Interesting, I did want to try ext4 just to verify, but I suppose that settles that. I am using arch and not silverblue however, I am using the flatpak version which used to work before after I moved the DS3 installation location.

**Comment by ghost** [comment]
> @Vovkiv Interesting, I did want to try ext4 just to verify, but I suppose that settles that. I am using arch and not silverblue however, I am using the flatpak version which used to work before after I moved the DS3 installation location.

@hebrewd i'm think it's more related to some lib, that steam uses in proton/wine to make DS3 works, or system configuration (for something) on different distros.

You also could try run different distro (like fedora workstation) from flash, install steam there (in flatpak) and try to run game. I'm more then sure, that game would work as expected (tho, you probably wouldn't be able to run save files form your arch system, because DS3 save files tied to system, so game would just ignore that save. It's stupid move from froms, but whatever)

**Comment by hebrewd** [comment]
> @hebrewd i'm think it's more related to some lib, that steam uses in proton/wine to make DS3 works, or system configuration (for something) on different distros.
> 
> You also could try run different distro (like fedora workstation) from flash, install steam there (in flatpak) and try to run game. I'm more then sure, that game would work as expected (tho, you probably wouldn't be able to run save files form your arch system, because DS3 save files tied to system, so game would just ignore that save. It's stupid move from froms, but whatever)

@Vovkiv yeah, I'm already trying on a new savefile either way to rule out that the save is the issue. Since the flatpak version also causes the issue, I think it is something on the lower end, I know that silverblue is an immutable distro at that. Does it use a different version of the linux kernel? or perhaps different mount options? I will install fedora and see if I can match the configuration to my arch install and iron out the issue.

**Comment by kisak-valve** [comment]
Proton has the `PROTON_SET_GAME_DRIVE` [runtime option](https://github.com/ValveSoftware/Proton/#runtime-config-options) to help games that are sensitive to long paths. It might be worth testing how the game behaves with the launch options set to `PROTON_SET_GAME_DRIVE=1 %command%`.

**Comment by hebrewd** [comment]
> Proton has the `PROTON_SET_GAME_DRIVE` [runtime option](https://github.com/ValveSoftware/Proton/#runtime-config-options) to help games that are sensitive to long paths. It might be worth testing how the game behaves with the launch options set to `PROTON_SET_GAME_DRIVE=1 %command%`.

@kisak-valve It solved it for me!

**Comment by ghost** [comment]
@kisak-valve @hebrewd
Huh, it's seems helped me too...
Tho, now it would be nice, why Fedora Silverblue need this, and Workstation no.
Because game installed on same drive, when i was changing distro, nothing changed. Steam was on same flatpak, everything was same.

**Comment by alasky17** [comment]
gamedrive should now be set automatically for this title.  Thank you for the reports!  If you run across any other games that need gamedrive (on any setup), please feel free to @ me :)

**Comment by R1chterScale** [comment]
Recently the Dark Souls 3 Lighting Engine mod released which provides a massive upgrade to the game's rendering pipeline. Sadly it doesn't seem to work through Proton, with it crashing on entering character creation. Proton log is attached. Also attached is the different crashlog with WineD3D 



[steam-374320.log.gz](https://github.com/user-attachments/files/21943833/steam-374320.log.gz)

[WineD3D.log](https://github.com/user-attachments/files/21944043/WineD3D.log)
